books/projects/x86isa/machine/instructions/arith-and-logic.lisp

```diff
@@ -295,6 +295,102 @@
 
     x86))
 
+
+
+(def-inst x86-adcx/adox
+
+  :parents (three-byte-opcodes)
+
+  :short "Operand Fetch and Execute for ADCX/ADOX"
+
+  :long "<h3>Op/En = RM: \[OP REG, R/M\] or \[OP G, E\]</h3>
+
+  <p>where @('G') is the destination operand and @('E') is the source
+  operand.  Note that @('G') stands for a general-purpose register
+  specified by the @('ModRM.reg') field, and @('E') stands for a
+  general-purpose register or memory operand specified by the
+  @('ModRM.r/m') field.</p>
+
+  <p>These instructions are essentially variations of ADC with different flag
+  behavior. ADCX sums the two operands and the CF flag (just like ADC), then
+  sets the CF flag according to unsigned overflow (just like ADC), but doesn't
+  modify any other flags. ADOX is the same as ADCX but uses/sets OF in place of
+  CF.</p>"
+
+
+  :returns (x86 x86p :hyp (x86p x86)
+                :hints (("Goal" :in-theory (e/d* ()
+                                                 (unsigned-byte-p
+                                                  signed-byte-p)))))
+
+  :modr/m t
+
+  :additional-args ((adcx booleanp))
+  
+  :body
+
+  (b* ((p2 (prefixes->seg prefixes))
+       (p4? (eql #.*addr-size-override* (prefixes->adr prefixes)))
+       (operand-size (if (logbitp *w* rex-byte) 8 4)) 
```

Comment:
Sorry, I just realized, this also doesn't account for 32-bit mode, where this is always 32-bit. Does `select-operand-size` not work here because the operand size shouldn't be less than 32-bits?

Reply:
I looked into this and I believe rex-byte will always be 0 if not in 64-bit mode -- get-prefixes is responsible for detecting the rex byte and it only does so if 64-bit mode. I think select-operand-size doesn't work unless we fake some stuff -- as you said it won't be less than 32 bits, plus the operand size for these instructions is dictated only by rex.w, not the default operand size and not the 0x66 prefix (which is a mandatory prefix for ADCX).

Reply:
I see, cool. Looks good then.